acceptance-tests/src/test/java/linea/plugin/acc/test/EcPairingLimitsTest.java

```diff
@@ -0,0 +1,225 @@
+      // nonce = nTransactions - 1, nTransactions - 2, ..., 0
+      int nonce = nTransactions - 1 - i;
+
+      // Craft the transaction data
+      final byte[] encodedCallEcPairing =
+          encodedCallEcPairing(
+              ecPairing,
+              ecPairingSender,
+              nonce,
+              Bytes.fromHexString(input.apply(i, nTransactions)));
+
+      // Send the transaction
+      final Web3j web3j = minerNode.nodeRequests().eth();
+      final EthSendTransaction resp =
+          web3j.ethSendRawTransaction(Numeric.toHexString(encodedCallEcPairing)).send();
+
+      // Store the transaction hash
+      txHashes[nonce] = resp.getTransactionHash();
+    }
+
+    // Transfer used as sentry to ensure a new block is mined
+    final Hash transferTxHash =
+        accountTransactions
+            .createTransfer(
+                accounts.getPrimaryBenefactor(),
+                accounts.getSecondaryBenefactor(),
+                1,
+                BigInteger.ONE) // nonce is 1 as primary benefactor also deploys the contract
+            .execute(minerNode.nodeRequests());
+    // Wait for the sentry to be mined
+    minerNode.verify(eth.expectSuccessfulTransactionReceipt(transferTxHash.toHexString()));
+
+    // Assert that all the transactions involving the EcPairing precompile, but the last one, were
+    // included in the same block
+    assertTransactionsMinedInSameBlock(
+        minerNode.nodeRequests().eth(), Arrays.asList(txHashes).subList(0, nTransactions - 1));
+
+    // Assert that the last transaction was included in another block
+    assertTransactionsMinedInSeparateBlocks(
+        minerNode.nodeRequests().eth(), List.of(txHashes[0], txHashes[nTransactions - 1]));
+
+    // Assert that the target string is contained in the block log
+    final String blockLog = getAndResetLog();
+    assertThat(blockLog).contains(target);
+  }
+
+  private static Stream<Arguments> ecPairingLimitsTestSource() {
+    List<Arguments> arguments = new ArrayList<>();
+
+    /*
+    This test case produces 17 transactions performing one ECPAIRING.
+    All ECPAIRING are well-formed and none of the points is ever the point at infinity,
+    leading in total to:
+    - 1 Miller loops
+    - 0 G2 membership tests
+    - 1 final exponentiations
+    per pairing.
+
+    In total:
+    - 17 Miller loops
+    - 0 G2 membership tests
+    - 17 final exponentiations
+
+    This final transaction exceeds the PRECOMPILE_ECPAIRING_FINAL_EXPONENTIATIONS and must be rejected.
+     */
+    arguments.add(
+        Arguments.of(
+            17, // 1 final exponentiation per transaction
+            (BiFunction<Integer, Integer, String>) (i, nTransactions) -> nonTrivial,
+            "Cumulated line count for module PRECOMPILE_ECPAIRING_FINAL_EXPONENTIATIONS=17 is above the limit 16, stopping selection"));
+
+    /*
+    This test case produces 9 transactions each performing 8 ECPAIRING's
```

Comment:
If this sentence is correct then one would expect (given that the pairing is nontrivial) 65 final exponentiations.